sentinelhub/sentinelhub_batch.py

```diff
@@ -0,0 +1,398 @@
+        return get_auth_json(url, request_type=RequestType.DELETE)
+
+    def start_analysis(self):
+        """ Starts analysis of a batch job request
+        """
+        return self._call_job('analyse')
+
+    def start_job(self):
+        """ Starts running a batch job
+        """
+        return self._call_job('start')
+
+    def cancel_job(self):
+        """ Cancels a batch job
+        """
+        return self._call_job('cancel')
+
+    def restart_job(self):
+        """ Restarts only those parts of a job that failed
+        """
+        return self._call_job('restartpartial')
+
+    def iter_tiles(self, status=None, **kwargs):
+        """ Iterate over info about batch request tiles
+
+        :param status: A filter to obtain only tiles with a certain status
+        :type status: str or None
+        :param kwargs: Any additional parameters to include in a request query
+        :return: An iterator over information about each tile
+        :rtype: Iterator[dict]
+        """
+        url = self._get_tiles_url()
+        params = _remove_undefined_params({
+            'status': status,
+            **kwargs
+        })
+        return _iter_pages(url, **params)
+
+    def get_tile(self, tile_id):
+        """ Provides information about a single batch request tile
+
+        :param tile_id: An ID of a tile
+        :type tile_id: int or None
+        :return: Information about a tile
+        :rtype: dict
+        """
+        url = self._get_tiles_url(tile_id=tile_id)
+        return get_auth_json(url)
+
+    def reprocess_tile(self, tile_id):
+        """ Reprocess a single failed tile
+
+        :param tile_id: An ID of a tile
+        :type tile_id: int or None
+        """
+        self._call_job(f'tiles/{tile_id}/restart')
+
+    def _parse_bounds_payload(self):
+        """ Parses bbox, geometry and crs from batch request payload. If bbox or geometry don't exist it returns None
+        instead.
+        """
+        bounds_definition = self.info['processRequest']['input']['bounds']
+        crs = CRS(bounds_definition['properties']['crs'].rsplit('/', 1)[-1])
+
+        return bounds_definition.get('bbox'), bounds_definition.get('geometry'), crs
+
+    def _call_job(self, endpoint_name):
+        """ Makes a POST request to the service that triggers a processing job
+        """
+        process_url = self._get_process_url(request_id=self.request_id, config=self.config)
+        url = f'{process_url}/{endpoint_name}'
+        return get_auth_json(url, request_type=RequestType.POST)
+
+    def _get_tiles_url(self, tile_id=None):
+        """ Creates an URL for tiles endpoint
+        """
+        process_url = self._get_process_url(config=self.config, request_id=self.request_id)
+        url = f'{process_url}/tiles'
+        if tile_id:
+            return f'{url}/{tile_id}'
+        return url
+
+    @staticmethod
+    def _get_process_url(config, request_id=None):
```

Comment:
These could be moved to some subclass or similar. `SentinelHubBatchEndpoints` perhaps?

Reply:
Hm, these static methods could become independent functions. But they are completely specific to `SentinelHubBatch` so it makes sense to have them as static methods.

I definitely would subclass for this.